coderd/tailnet.go

```diff
@@ -351,13 +354,43 @@ func (s *ServerTailnet) ReverseProxy(targetURL, dashboardURL *url.URL, agentID u
 	tgt.Host = net.JoinHostPort(tailnet.IPFromUUID(agentID).String(), port)
 
 	proxy := httputil.NewSingleHostReverseProxy(&tgt)
-	proxy.ErrorHandler = func(w http.ResponseWriter, r *http.Request, err error) {
+	proxy.ErrorHandler = func(w http.ResponseWriter, r *http.Request, theErr error) {
+		var (
+			desc         = "Failed to proxy request to application: " + theErr.Error()
+			additional   = ""
+			switchLink   = ""
+			switchTarget = ""
+		)
+
+		if strings.Contains(theErr.Error(), "tls:") {
```

Comment:
The error check should test the type and the actual message too. I think we only want to show this if the error is "first record does not look like a TLS handshake".

Golang sets the error type to [`tls.RecordHeaderError`](https://pkg.go.dev/crypto/tls#RecordHeaderError) [source](https://cs.opensource.google/go/go/+/master:src/crypto/tls/conn.go;l=669;drc=1e12eab8705d1d8d7472be9147a39caa1c8380db), and you could match the specific string on the inner `Msg` field after doing `xerrors.As`

Reply:
Should probably also match [`http.ErrSchemeMismatch`](https://pkg.go.dev/net/http#ErrSchemeMismatch) using `xerrors.Is`